accord-core/src/main/java/accord/impl/progresslog/WaitingState.java

```diff
@@ -0,0 +1,660 @@
+    {
+        fetchCallback(FetchRoute, safeStore, safeCommand, owner, txnId, blockedUntil, fetchResult, fail);
+    }
+
+    static void fetchCallback(SafeCommandStore safeStore, SafeCommand safeCommand, DefaultProgressLog owner, TxnId txnId, BlockedUntil blockedUntil, FetchData.FetchResult fetchResult, Throwable fail)
+    {
+        fetchCallback(Fetch, safeStore, safeCommand, owner, txnId, blockedUntil, fetchResult, fail);
+    }
+
+    static void fetchCallback(CallbackKind kind, SafeCommandStore safeStore, SafeCommand safeCommand, DefaultProgressLog owner, TxnId txnId, BlockedUntil blockedUntil, FetchData.FetchResult fetchResult, Throwable fail)
+    {
+        Invariants.checkState(fetchResult != null || fail != null);
+        Unseekables<?> ready = fetchResult == null ? null : fetchResult.achievedTarget;
+        Unseekables<?> notReady = fetchResult == null ? null : fetchResult.didNotAchieveTarget;
+        BlockedUntil upgrade = fetchResult == null ? null : BlockedUntil.forSaveStatus(fetchResult.achieved.propagatesSaveStatus());
+        awaitOrFetchCallback(kind, safeStore, safeCommand, owner, txnId, blockedUntil, ready, notReady, upgrade, fail);
+    }
+
+    static void synchronousAwaitHomeCallback(SafeCommandStore safeStore, SafeCommand safeCommand, DefaultProgressLog owner, TxnId txnId, BlockedUntil blockedUntil, AsynchronousAwait.SynchronousResult awaitResult, Throwable fail)
+    {
+        synchronousAwaitCallback(AwaitHome, safeStore, safeCommand, owner, txnId, blockedUntil, awaitResult, fail);
+    }
+
+    static void synchronousAwaitSliceCallback(SafeCommandStore safeStore, SafeCommand safeCommand, DefaultProgressLog owner, TxnId txnId, BlockedUntil blockedUntil, AsynchronousAwait.SynchronousResult awaitResult, Throwable fail)
+    {
+        synchronousAwaitCallback(AwaitSlice, safeStore, safeCommand, owner, txnId, blockedUntil, awaitResult, fail);
+    }
+
+    static void synchronousAwaitCallback(CallbackKind kind, SafeCommandStore safeStore, SafeCommand safeCommand, DefaultProgressLog owner, TxnId txnId, BlockedUntil blockedUntil, AsynchronousAwait.SynchronousResult awaitResult, Throwable fail)
+    {
+        Unseekables<?> ready = awaitResult == null ? null : awaitResult.ready;
+        Unseekables<?> notReady = awaitResult == null ? null : awaitResult.notReady;
+        // TODO (desired): extract "upgrade" info from AsynchronousAwait
+        awaitOrFetchCallback(kind, safeStore, safeCommand, owner, txnId, blockedUntil, ready, notReady, null, fail);
+    }
+    
+    void asynchronousAwaitCallback(DefaultProgressLog owner, SafeCommandStore safeStore, SaveStatus newStatus, Node.Id from, int callbackId)
+    {
+        if ((callbackId & 1) != 1)
+            return;
+
+        BlockedUntil blockedUntil = blockedUntil();
+        if (callbackId == Integer.MAX_VALUE)
```

Comment:
Can you use a named constant here instead of Integer.MAX_VALUE?

Reply:
yep